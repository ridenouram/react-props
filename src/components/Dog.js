import React from 'react';
import PropTypes from 'prop-types';

function Dog({ name, age, weight }) {
  return (
    <dl>
      <dt>Name</dt>
      <dd>{name}</dd>
  
      <dt>Age</dt>
      <dd>{age}</dd>
  
      <dt>Weight</dt>
      <dd>{weight ? weight : 'unknown'}</dd>
    </dl>
  );
}

Dog.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  weight: PropTypes.string
}

export default Dog;

