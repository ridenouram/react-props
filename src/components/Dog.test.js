import React from 'react';
import { shallow } from 'enzyme';
import Dog from './Dog';

describe('Header Component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Dog />);
    expect(wrapper).toMatchSnapshot();
  });
});
