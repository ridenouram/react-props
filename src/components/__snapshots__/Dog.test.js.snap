// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches a snapshot 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd />
  <dt>
    Age
  </dt>
  <dd />
  <dt>
    Weight
  </dt>
  <dd>
    unknown
  </dd>
</dl>
`;
